src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs

```diff
@@ -340,19 +352,25 @@ private static void HookOrProcessResult(HttpWebRequest request)
         if (endCalledAccessor.Invoke(readAsyncContext) || readAsyncContext.CompletedSynchronously)
         {
             // We need to process the result directly because the read callback has already fired. Force a copy because response has likely already been disposed.
-            ProcessResult(readAsyncContext, null, writeAsyncContextCallback.Activity, resultAccessor(readAsyncContext), true);
+            ProcessResult(readAsyncContext, null, writeAsyncContextCallback.Activity, resultAccessor(readAsyncContext), true, request, writeAsyncContextCallback.StartTimestamp);
             return;
         }
 
         // Hook into the result callback if it hasn't already fired.
-        AsyncCallbackWrapper callback = new AsyncCallbackWrapper(writeAsyncContextCallback.Request, writeAsyncContextCallback.Activity, asyncCallbackAccessor(readAsyncContext));
+        AsyncCallbackWrapper callback = new AsyncCallbackWrapper(writeAsyncContextCallback.Request, writeAsyncContextCallback.Activity, asyncCallbackAccessor(readAsyncContext), Stopwatch.GetTimestamp());
         asyncCallbackModifier(readAsyncContext, callback.AsyncCallback);
     }
 
-    private static void ProcessResult(IAsyncResult asyncResult, AsyncCallback asyncCallback, Activity activity, object result, bool forceResponseCopy)
+    private static void ProcessResult(IAsyncResult asyncResult, AsyncCallback asyncCallback, Activity activity, object result, bool forceResponseCopy, HttpWebRequest request, long startTimestamp)
     {
+        HttpStatusCode? httpStatusCode = null;
+
+        // Activity may be null if we are not tracing in these cases:
+        // 1. No listeners
+        // 2. Request was filtered out
+        // 3. Request was not sampled
         // We could be executing on a different thread now so restore the activity if needed.
-        if (Activity.Current != activity)
+        if (activity != null && Activity.Current != activity)
```

Comment:
I think some of the logic in the block below this will blow up for `null` activity like `AddResponseTags`. Maybe surround all of the tracing logic with an `if (activity != null)` check or refactor it into a method you can call when `activity != null`?

Reply:
`AddResponseTags` and `AddExceptionTags` both have added null checks for activity. Can move these null checks to the calling method easily, but my general approach was to put the checks in any method called multiple times, and add conditionals if a method was called once in the request/response flow.

Reply:
Let's change the `null` checks in `AddResponseTags` and `AddExceptionTags` into `Debug.Assert`s to help catch any potential mistakes and then make sure `ProcessResult` bypasses all of that logic. One of the goals we have with instrumentation is for them to be ~transparent to the process when disabled so reducing operations (give back more CPU cycles) would be great for that.